homeassistant/components/teslemetry/climate.py

```diff
@@ -102,6 +103,10 @@ def _async_update_attrs(self) -> None:
         else:
             self._attr_hvac_mode = HVACMode.OFF
 
+        # If not scoped, prevent the user from changing the HVAC mode by making it the only option
+        if self._attr_hvac_mode and not self.scoped:
+            self._attr_hvac_modes = [self._attr_hvac_mode]
```

Comment:
Do you need to handle the `self._attr_hvac_mode = None` case that resets `_attr_hvac_modes` to be empty/none?

Reply:
That case only occurs before the coordinator has successfully polled the data, and as such I set the value of HVAC modes to `[]` at startup.

Since this code block ensures HVAC mode is truthy it won't put None in HVAC Modes.

Reply:
I'm not seeing the hvac modes being set to an empty list at start. Is this up to date?

Reply:
Good point, that must be the other PR I did for Tesla Fleet, I'll make that change.

Reply:
Ive added this in and refactored the init so it works (similar to what I did for Tesla Fleet)